ledger/participant-integration-api/src/main/scala/platform/apiserver/RateLimitingInterceptor.scala

```diff
@@ -139,21 +123,19 @@ private[apiserver] final class RateLimitingInterceptor(
     memoryMxBean.gc()
   }
 
-  private def metricOverloaded(
-      fullMethodName: String,
+  private def queueUnderLimit(
       count: InstrumentedCount,
       limit: Int,
-  ): Either[String, Unit] = {
+  ): Either[DamlError, Unit] = {
     val queued = count.queueSize
     if (queued > limit) {
-      val rpcStatus =
-        s"""
+
+      val damlError = QueueSizeOverLimit.Rejection(s"""
            | The ${count.name} queue size ($queued) has exceeded the maximum ($limit).
-           | The rejected call was $fullMethodName.
```

Comment:
If you decide to only log from error instantiation, I would leave the `fullMethodName` as part of the method name.